test/unit/plugins/validate-semantic/oas3/refs.js

```diff
@@ -206,4 +206,82 @@ describe("validation plugin - semantic - oas3 refs", () => {
       return expectNoErrorsOrWarnings(spec)
     })
   })
+
+  describe("$refs for requestbody schemas must reference a schema by position", () => {
+    it("should return an error when a requestBody schema incorrectly references a local component schema", () => {
+      const spec = {
+        openapi: "3.0.0",
+        paths: {
+          "/foo": {
+            post: {
+              requestBody: {
+                content: {
+                  "application/json": {
+                    schema: {
+                      $ref: "#/components/requestbodies/Foo"
+                    }
+                  }
+                }
+              }
+            }
+          }
+        },
+        components: {
+          schemas: {
+            Foo: {
+              type: "string"
+            }
+          }
+        }
+      }
+
+      return validateHelper(spec)
+        .then(system => {
+          const allErrors = system.errSelectors.allErrors().toJS()
+          const allSemanticErrors = allErrors.filter(err => err.source === "semantic")
+          
+          expect(allSemanticErrors.length).toEqual(3)
+          
+          const firstError = allSemanticErrors[2]
+          
+          expect(firstError.message).toEqual(`requestBody schemas $refs must point to a position where a schema can be legally placed`)
+          expect(firstError.path).toEqual(["paths", "/foo", "post", "requestBody", "content", "application/json", "schema", "$ref"])
+
+        })
+    })
+
+    it("should not return an error when a requestBody schema references a local component schema", () => {
+      const spec = {
+        openapi: "3.0.0",
+        paths: {
+          "/foo": {
+            post: {
+              requestBody: {
+                content: {
+                  "application/json": {
+                    schema: {
+                      $ref: "#/components/schemas/Foo"
+                    }
+                  }
+                }
+              }
+            }
+          }
+        },
+        components: {
+          schemas: {
+            Foo: {
+              type: "string"
+            }
+          }
+        }
+      }
+
+      return validateHelper(spec)
+        .then(system => {
+          const allErrors = system.errSelectors.allErrors().toJS()
+          expect(allErrors.length).toEqual(0)
+        })
+    })
+  })
```

Comment:
**Change request**: please add a test case that ensures local operation $refs aren't caught by this validator, e.g. this should have no problems:

```yaml
openapi: 3.0.0
info:
  title: Enum value validation against schema
  version: 1.0.0
paths: 
  /foo:
    post:
      responses:
        200:
          description: ok
      requestBody:
        content:
          application/json:
            schema:
              type: string
    put:
      responses:
        200:
          description: ok
      requestBody:
        content:
          application/json:
            schema:
              $ref: "#/paths/~1foo/post/requestBody/content/application~1json/schema"
```

Reply:
Test added!